contracts/ArrowToken.sol

```diff
@@ -1,29 +1,39 @@
-pragma solidity ^0.8.11;
-
 // SPDX-License-Identifier: MIT
+pragma solidity ^0.8.11;
 
-import "@openzeppelin/contracts/access/Ownable.sol";
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 /**
-    @title Arrow Token implementation
- */
-contract ArrowToken is ERC20, Ownable {
+    @title Arrow Token Implementation
+*/
+contract ArrowToken is
+    Initializable,
+    ERC20Upgradeable,
+    OwnableUpgradeable,
+    UUPSUpgradeable
+{
+    // implement the UUPS interface
+    function _authorizeUpgrade(address) internal override onlyOwner {}
 
-    /// @notice This function assigns msg.sender as the token admin and mints them initial supply
-    constructor(uint256 initialSupply, string memory _name, string memory _symbol)
-        ERC20(_name, _symbol)
-    {
-        _mint(msg.sender, initialSupply);
-    }
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() initializer {}
+
+    /** 
+        @notice Mint an `_initialSupply` amount of tokens upon contract creation
+        @dev If we were to upgrade the initializer by introducing new variables, we need to create a normal function like `initializeV2` (https://forum.openzeppelin.com/t/uups-proxies-tutorial-solidity-javascript/7786/53), and make sure to include all existing variables from the current version of the contract to avoid storage collision
+    */
+    function initialize(uint256 _initialSupply, string memory _name, string memory _symbol) public initializer {
+        __ERC20_init(_name, _symbol);
+        __Ownable_init();
+        __UUPSUpgradeable_init();
 
-    /**
-        Mints new tokens.
+        _mint(msg.sender, _initialSupply);
+    }
 
-        Only accepts calls from the contract owner.
-     */
-    function mint(address to, uint256 amount) public onlyOwner
-    {
-        _mint(to, amount);
+    function mint(address _to, uint256 _amount) public onlyOwner {
```

Comment:
Can you add the documentation back to this function? There's no need to remove it.